test/archive/har.d

```diff
@@ -0,0 +1,198 @@
+        }
+    
+        if (!exists(harFilename))
+        {
+            infoFile.writefln("Error: %s does not exist", harFilename.formatFile);
+            return No.success;
+        }
+        auto harFile = File(harFilename, "r");
+    
+        if (exists(outputDir))
+        {
+            if (!isDir(outputDir))
+            {
+                infoFile.writefln("Error: output directory %s already exists and is not a directory", outputDir.formatDir);
+                return No.success;
+            }
+            // Maybe a warning if directory is not empty or something
+            if (verbose)
+                infoFile.writefln("output directory %s already exists", outputDir.formatDir);
+        }
+        else
+        {
+            if (verbose)
+                infoFile.writefln("mkdir %s", outputDir.formatDir);
+            mkdirRecurse(outputDir);
+        }
+    
+        auto lineRange = harFile.byLine(Yes.keepTerminator);
+        if (lineRange.empty)
+        {
+            infoFile.writefln("%s(1): Error: file is empty", harFilename);
+            return No.success;
+        }
+        string delimiter;
+        auto line = lineRange.front;
+        {
+            auto spaceIndex = line.indexOf(' ');
+            if (spaceIndex < 0)
+            {
+                infoFile.writefln("%s(1) Error: first line does not contain a space", harFilename);
+                return No.success;
+            }
+            delimiter = line[0 .. spaceIndex + 1].idup;
+        }
+    
+    LfileLoop:
+        for (;;)
+        {
+            // At this point, line starts with the delimiter
+            auto fileInfoString = line[delimiter.length .. $];
+    
+            // For now this will just be the filename
+            auto fileName = fileInfoString.stripRight;
+            {
+                auto dir = dirName(fileName);
+                if (dir.length > 0)
+                {
+                    auto fullDir = buildPath(outputDir, dir);
+                    if (!extractMkdir(fullDir))
+                        return No.success;
+                }
+            }
+            auto fullFileName = buildPath(outputDir, fileName);
+            infoFile.writefln("creating %s", fullFileName.formatFile);
+            extractedFilenames.put(fullFileName.idup);
+            {
+                auto currentOutputFile = File(fullFileName, "w");
+                scope(exit) currentOutputFile.close();
+                for (;;)
+                {
+                    lineRange.popFront();
+                    if (lineRange.empty)
+                        break LfileLoop;
+                    line = lineRange.front;
+                    if (line.startsWith(delimiter))
+                        break;
+                    currentOutputFile.write(line);
+                }
+            }
+        }
+        return Yes.success;
+    }
+}
+
+@property auto formatDir(const(char)[] dir)
```

Comment:
The `@property` seems unnecessary here.

Reply:
What criteria do you use to determine when to use `@property`?

Reply:
Never.
At the moment it has nearly no effect. It just enforces that a member function without argument is called without parenthesis. https://github.com/dlang/DIPs/pull/97 might fix this, but it's still a long journey until then.

See also: https://dlang.org/spec/function.html#property-functions

Reply:
I currently use it as a means to indicate that a "free function" (non-member function) should always be called with UFCS with no arguments.  A way to indicate it should always be called like accessing a "member field/property" indicating it doesn't do much but instead is meant to expose data from the object (which is the first argument).

Reply:
@wilzbach doesn't seem to have much affect at all: https://run.dlang.io/is/8WPlhI.

Reply:
Then it's even more useless than I thought :/

Reply:
It did have a functionality with the `-property` flag. But that's now deprecated and has now affect.

Reply:
ok, well looks like no one thinks my use has much value so I've removed `@property`